nsdb-common/src/main/scala/io/radicalbit/nsdb/common/statement/SQLStatement.scala

```diff
@@ -150,45 +149,78 @@ case object LessOrEqualToOperator    extends ComparisonOperator
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(
+  Array(
+    new JsonSubTypes.Type(value = classOf[CountAggregation], name = "CountAggregation"),
+    new JsonSubTypes.Type(value = classOf[CountDistinctAggregation], name = "CountDistinctAggregation"),
+    new JsonSubTypes.Type(value = classOf[MaxAggregation], name = "MaxAggregation"),
+    new JsonSubTypes.Type(value = classOf[MinAggregation], name = "MinAggregation"),
+    new JsonSubTypes.Type(value = classOf[SumAggregation], name = "SumAggregation"),
+    new JsonSubTypes.Type(value = classOf[FirstAggregation], name = "FirstAggregation"),
+    new JsonSubTypes.Type(value = classOf[LastAggregation], name = "LastAggregation"),
+  ))
 sealed trait PrimaryAggregation extends Aggregation
 
 /**
   * Aggregation that is derived from others.
   * e.g. average is derived from count and sum.
   */
-@JsonSerialize(using = classOf[DerivedAggregationJsonSerializer])
-@JsonDeserialize(using = classOf[DerivedAggregationJsonDeserializer])
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(
+  Array(
+    new JsonSubTypes.Type(value = classOf[AvgAggregation], name = "AvgAggregation")
+  ))
 sealed trait DerivedAggregation extends Aggregation {
   def primaryAggregationsRequired: List[Aggregation with PrimaryAggregation]
 }
 
-case object CountAggregation         extends GlobalAggregation with PrimaryAggregation
-case object CountDistinctAggregation extends GlobalAggregation with PrimaryAggregation
-case object MinAggregation           extends GlobalAggregation with PrimaryAggregation
-case object MaxAggregation           extends Aggregation with PrimaryAggregation
-case object FirstAggregation         extends Aggregation with PrimaryAggregation
-case object LastAggregation          extends Aggregation with PrimaryAggregation
-case object SumAggregation           extends Aggregation with PrimaryAggregation
-case object AvgAggregation extends GlobalAggregation with DerivedAggregation {
+case class CountAggregation(override val fieldName: String)         extends GlobalAggregation with PrimaryAggregation
+case class CountDistinctAggregation(override val fieldName: String) extends GlobalAggregation with PrimaryAggregation
+case class MaxAggregation(override val fieldName: String)           extends Aggregation with PrimaryAggregation
+case class MinAggregation(override val fieldName: String)           extends GlobalAggregation with PrimaryAggregation
+case class FirstAggregation(override val fieldName: String)         extends Aggregation with PrimaryAggregation
+case class LastAggregation(override val fieldName: String)          extends Aggregation with PrimaryAggregation
+case class SumAggregation(override val fieldName: String)           extends Aggregation with PrimaryAggregation
+case class AvgAggregation(override val fieldName: String) extends GlobalAggregation with DerivedAggregation {
```

Comment:
is `override val` mandatory?

Reply:
No, but I'd prefer to keep it, just to explicitly point out that it's an inherited field